src/datadog/id_generator.cpp

```diff
@@ -15,11 +16,17 @@ class DefaultIDGenerator : public IDGenerator {
   explicit DefaultIDGenerator(bool trace_id_128_bit)
       : trace_id_128_bit_(trace_id_128_bit) {}
 
-  TraceID trace_id() const override {
+  TraceID trace_id(const TimePoint& start) const override {
     TraceID result;
     result.low = random_uint64();
     if (trace_id_128_bit_) {
-      result.high = random_uint64();
+      // Highest 32 bits contain a unix timestamp (the trace start time).
+      const auto since_epoch = start.wall.time_since_epoch();
+      const auto seconds =
+          std::chrono::duration_cast<std::chrono::seconds>(since_epoch).count();
+      // The farthest we'll go back is the unix epoch.
+      const std::uint64_t unsigned_seconds = seconds < 0 ? 0 : seconds;
```

Comment:
I agree it's not going to happen. But why 1 and not 0?

My understanding of unix time is that is the number of seconds elapsed since 00:00:00 UTC on 1 January 1970.

Hm, wait...
```console
david@ein:~/Downloads$ python
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import datetime     
>>> datetime.datetime.fromtimestamp(0)
datetime.datetime(1969, 12, 31, 19, 0)
```
Ah, that's a time zone thing.

Anyway, the reason I have a cutoff here is to prevent the rollover to 18446744073709551615 at -1 (which is not a valid unix timestamp, but would correspond to a date over 584 trillion years in the future -- build software to last!).